include/splinterdb/splinterdb.h

```diff
@@ -325,4 +323,8 @@ splinterdb_stats_print_lookup(const splinterdb *kvs);
 void
 splinterdb_stats_reset(splinterdb *kvs);
 
+// TRUE if Splinter's task system was configured to use background threads.
+bool
+splinterdb_task_system_uses_bg_threads(splinterdb *kvs);
```

Comment:
Externalized existing boolean fn, to be used as a testing hook.

Reply:
Why can't someone just get this from the splinter config?

Reply:
@rtjohnso - Test code uses this to cross-check whether the background threads were, indeed, created [or not] for a specific config params for bg-threads.

For example, if you specify normal_bg_threads=1 and memtable_bg_threads=0, with this fix, no background threads are created.

Test code re-confirms this behaviour. See [this test case, test_splinterdb_created_wo_background_threads] in task_system_test.c

Reply:
If you remove that "silently configure" bit and instead make it a hard error, would that mean your tests wouldn't require this function?

Reply:
@rosenhouse - This is a testability viewpoint.

If we make it a hard-error when one of the 2 config-params is zero, then, we are [implicitly] assuming that for non-zero config params, that the bg-threads are created.

If I withdraw this lookup function, tests do not have a way of "verifying" that the bg-threads _were_ created as claimed.

In my book, even if I went to the hard-error route, I'd like to retain this interface as it provides a way to externalize the state of Splinter's sub-systems. Using it for testing would be one such immediate consumer. (Splinter, I think, solely lacks externally usable reporting interfaces and I kind see this a baby-step towards improving that situation.)

Reply:
Maybe we need to discuss this on zoom, because I think I'm seeing it differently.  A user ought to be able to easily predict when background threads will be created or not, based entirely on the config they passed in.

I think "reporting" or "metrics" interfaces are a good idea in general, I just don't think we should start designing that feature in this PR -- it should begin with a design doc.

Reply:
Agreed to ... Withdraw this interface as it is in public .h file.